crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/CheckDiagnosticsCommand.java

```diff
@@ -66,18 +66,19 @@ public CallStatus run() {
                     String errorDetails = entry.getKey().toString() + " " + entry.getValue().getDetails();
                     Error.push("diagnosticCheck failed! " + errorDetails);
                     LOGGER.debug("diagnosticCheck failed!", errorDetails);
-                    return new CallStatus(Error.get(RestCrudConstants.ERR_REST_CHECK_DIAGNOSTICS, errorDetails));
+                    healthCheckResults.put(entry.getKey(), entry.getValue());
+                    healthCheckResponse = Response.Status.SERVICE_UNAVAILABLE;
```

Comment:
Could probably simplify this a touch by putting healthCheckResults.put(entry.getKey(), entry.getValue()); outside of the conditional since it's in both branches.